plugins/cookieauth.js

```diff
@@ -32,7 +32,9 @@ class CookiePlugin extends BasePlugin {
     let sessionUrl = new u.URL(cfg.serverUrl);
     sessionUrl.pathname = '/_session';
     if (!sessionUrl.username || !sessionUrl.password) {
```

Comment:
I think we could add another flag to this check e.g. `errorOnNoCreds` so that when the cookie plugin is explicitly added by a user, but has no credentials that flag would default to `true` and they would receive an error for the misconfiguration. When we add the plugin by default we can set it to `false` in the configuration and allow the no auth case to not error.